src/core/server/saved_objects/migrations/core/document_migrator.ts

```diff
@@ -279,7 +279,7 @@ function props(doc: SavedObjectUnsanitizedDoc) {
  */
 function propVersion(doc: SavedObjectUnsanitizedDoc | ActiveMigrations, prop: string) {
   return (
-    (doc[prop] && doc[prop].latestVersion) ||
+    ((doc as any)[prop] && (doc as any)[prop].latestVersion) ||
     (doc.migrationVersion && (doc as any).migrationVersion[prop])
```

Comment:
Wonder if it's worth to introduce an internal type that keeps this `[rootProp: string]: any` ?

Reply:
I considered it, but it felt like we're losing type safety instead of increasing it. Core doesn't usually touch any of the root properties except the ones that core itself defines (updated_at namespaces), this section that performs the migration is the only exception where Core basically touches data owned by a plugin. So it felt like we could rather have `any` override here with slightly more type safety everywhere else.

Reply:
SGTM